flexget/plugins/sites/newznab.py

```diff
@@ -1,125 +1,599 @@
+    'tv/foreign': 5020,
+    'tv/sd': 5030,
+    'tv/hd': 5040,
+    'tv/uhd': 5045,  # unofficial
+    'tv/other': 5050,
+    'tv/sport': 5060,
+    'tv/anime': 5070,
+    'tv/documentary': 5080,  # unofficial
+    # XXX
+    'xxx': 6000,
+    'xxx/dvd': 6010,
+    'xxx/wmv': 6020,
+    'xxx/xvid': 6030,
+    'xxx/x264': 6040,
+    'xxx/uhd': 6045,  # unofficial
+    'xxx/pack': 6050,
+    'xxx/imageset': 6060,
+    'xxx/packs': 6070,  # unofficial
+    'xxx/sd': 6080,  # unofficial
+    'xxx/web-dl': 6090,  # unofficial
+    # Misc
+    'misc': 7000,
+    'misc/misc': 7010,
+    'misc/ebook': 7020,
+    'misc/comics': 7030,
+    # Unknown
+    'not-determined': 7900
+}
+
+field_map = {
+    'title': 'title',
+    'url': 'link',
+    'newznab_pubdate': lambda xml: value_to_naive_utc(find_value(['updated_parsed', 'updated'], xml)),
+    'newznab_grabs': lambda xml: value_to_int(find_value('newznab_grabs.value', xml)),
+    'newznab_size': lambda xml: value_to_int(find_value('newznab_size.value', xml)),
+    'newznab_files': lambda xml: value_to_int(find_value('newznab_files.value', xml)),
+    'newznab_usenet_date': lambda xml: value_to_naive_utc(find_value('newznab_usenetdate.value', xml)),
+    'newznab_password': lambda xml: value_to_int(find_value('newznab_password.value', xml)),
+    'newznab_guid': lambda xml: re.split(r'[/=]', find_value(['id', 'guid'], xml, default='')).pop(),
+    'newznab_usenet_group': 'newznab_group.value',
+    'newznab_tvairdate': lambda xml: value_to_naive_utc(find_value('newznab_tvairdate.value', xml)),
+    'newznab_season':
+        lambda xml: value_to_int(find_value('newznab_season.value', xml, regex=r'[Ss]([0-9]{1,2})')),
+    'newznab_episode':
+        lambda xml: value_to_int(find_value('newznab_episode.value', xml, regex=r'[Ee]([0-9]{1,3})')),
+    'newznab_hydra_indexer_name': 'newznab_hydraindexername.value',
+    'newznab_hydra_indexer_host': 'newznab_hydraindexerhost.value',
+    'newznab_hydra_indexer_score': lambda xml: value_to_int(find_value('newznab_hydraindexerscore.value', xml))
+}
+
+field_validation_list = [
+    'title',
+    'url'
+]
+
+PLUGIN_LOOKUP_MAP = {
+    'tvdb': 'thetvdb_lookup',
+    'trakt': 'trakt_lookup',
+    'tvmaze': 'tvmaze_lookup',
+    'imdb': 'imdb_lookup',
+    'tmdb': 'tmdb_lookup'
+}
+
+NAMESPACE_NAME = 'newznab'
+NAMESPACE_URL = 'http://www.newznab.com/DTD/2010/feeds/attributes/'
+NAMESPACE_TAGNAME = 'attr'
+ENCLOSURE_TYPE = 'application/x-nzb'
+
+
+# list utils
+def list_insert_unique(in_list, idx, element, caseinsensitive=False):
+    if not isinstance(in_list, list) or element is None:
+        return
+    if isinstance(element, str):
+        if element.isspace() or element in in_list:
+            return
+        elif caseinsensitive is True and element.lower() in in_list:
+            return
+    elif element in in_list:
+        return
+    in_list.insert(idx, element)
+
+
+def list_append_unique(in_list, element, caseinsensitive=False):
```

Comment:
Yes, but you never use the default from what I can see. Maybe the function should really just be case insensitive then.